src/installer.rs

```diff
@@ -60,91 +63,135 @@ impl Display for LoaderVersion {
+    Ok(get("https://meta.quiltmc.org/v3/versions/loader")
+        .await?
+        .json()
+        .await?)
 }
 
-#[derive(Serialize, Deserialize)]
-struct LaunchProfiles {
-    profiles: HashMap<String, serde_json::Value>,
-    settings: serde_json::Value,
-    version: u32
+/// `Deserialize` is not implemented for a reason
+///
+/// DO NOT deserialise `launcher_profiles.json` into this incomplete struct and write it back as it will cause **data loss**
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Profile {
+    name: String,
+    #[serde(rename = "type")]
+    profile_type: String,
+    created: DateTime<Utc>,
+    last_version_id: String,
+    icon: String,
 }
 
 pub async fn install_client(args: ClientInstallation) -> Result<()> {
     println!("Installing client: {:#?}", args);
 
     // Verify install location
     if !args.install_location.exists() {
-        return Err(anyhow!("Target directory doesn't exist: {:?}", args.install_location));
+        return Err(anyhow!(
+            "Target directory doesn't exist: {:?}",
+            args.install_location
+        ));
     }
 
     // Resolve profile directory
-    let profile_name = format!("quilt-loader-{}-{}", args.loader_version.version, args.minecraft_version.version);
-    let mut profile_dir = args.install_location.clone();
-    profile_dir.push("versions");
-    profile_dir.push(&profile_name);
+    let profile_name = format!(
+        "quilt-loader-{}-{}",
+        args.loader_version.version, args.minecraft_version.version
+    );
+    let profile_dir = args.install_location.join("versions").join(&profile_name);
 
-    // Delete existing profile
     if profile_dir.exists() {
-        std::fs::remove_dir_all(&profile_dir)?;
+        // Delete existing profile
+        remove_dir_all(&profile_dir)?;
     }
+    create_dir_all(&profile_dir)?;
 
-    // Create directory
-    std::fs::create_dir_all(&profile_dir)?;
+    // An empty jar file to make the vanilla launcher happy
+    File::create(profile_dir.join(profile_name.clone() + ".jar"))?;
 
-    // NOTE: This is an empty jar file to make the vanilla launcher happy
-    let mut jar_path = profile_dir.clone();
-    jar_path.push(format!("{}.jar", &profile_name));
-    File::create(jar_path)?;
-    
     // Create launch json
-    let mut json_path = profile_dir.clone();
-    json_path.push(format!("{}.json", &profile_name));
-    let mut file = File::create(json_path)?;
+    let mut file = File::create(profile_dir.join(profile_name.clone() + ".json"))?;
 
     // Download launch json
-    let response = reqwest::get(format!("https://meta.quiltmc.org/v3/versions/loader/{}/{}/profile/json", &args.minecraft_version.version, &args.loader_version.version)).await?.text().await?;
+    let mut response = get(format!(
+        "https://meta.quiltmc.org/v3/versions/loader/{}/{}/profile/json",
+        &args.minecraft_version.version, &args.loader_version.version
+    ))
+    .await?
+    .text()
+    .await?;
```

Comment:
~~One line!~~ Nope, I'm actually glad that this one is multiple lines now :P

Reply:
Could've even been
```rust
fn function() -> Result<…> {
    get("https://meta.quiltmc.org/v3/versions/loader")
        .await?
        .json()
        .await
}
```
but the error wouldn't get converted.